src/data/finset/basic.lean

```diff
@@ -166,6 +166,12 @@ theorem mem_def {a : α} {s : finset α} : a ∈ s ↔ a ∈ s.1 := iff.rfl
 instance decidable_mem [h : decidable_eq α] (a : α) (s : finset α) : decidable (a ∈ s) :=
 multiset.decidable_mem _ _
 
+instance decidable_forall_mem {P : α → Prop} [decidable_pred P] (s : finset α) :
+  decidable (∀ a ∈ s, P a) := s.val.decidable_forall_mem
+
+instance decidable_exists_mem {P : α → Prop} [decidable_pred P] (s : finset α) :
+  decidable (∃ a ∈ s, P a) := s.val.decidable_exists_mem
```

Comment:
These already exist as `finset.decidable_dforall_finset` and `finset.decidable_dexists_finset` (except that yours are non-dependent functions).

Reply:
Why do we have the non-dependent list version at all?

Reply:
No idea. Maybe they were added for unification? It seems like nothing breaks if you remove them.

Reply:
No idea. Maybe they were added for unification? It seems like nothing breaks if you remove them.

Reply:
No idea. Maybe they were added for unification? It seems like nothing breaks if you remove them.